File: src/main/java/no/nav/fo/veilarbregistrering/registrering/tilstand/RegistreringTilstandService.java
package no.nav.fo.veilarbregistrering.registrering.tilstand;

import no.nav.fo.veilarbregistrering.registrering.tilstand.resources.RegistreringTilstandDto;

import java.util.List;

public class RegistreringTilstandService {

    private final RegistreringTilstandRepository registreringTilstandRepository;

    public RegistreringTilstandService(RegistreringTilstandRepository registreringTilstandRepository) {
        this.registreringTilstandRepository = registreringTilstandRepository;
    }

    public void oppdaterRegistreringTilstand(RegistreringTilstandDto registreringTilstandDto) {
        RegistreringTilstand original = registreringTilstandRepository.hentRegistreringTilstand(registreringTilstandDto.getId());
        RegistreringTilstand oppdatert = original.oppdaterStatus(registreringTilstandDto.getStatus());
        registreringTilstandRepository.oppdater(oppdatert);
    }

    public List<RegistreringTilstand> finnRegistreringTilstandMed(Status status) {
        return registreringTilstandRepository.finnRegistreringTilstandMed(status);
    }

}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/registrering/formidling/RegistreringTilstandService.kt
package no.nav.fo.veilarbregistrering.registrering.formidling

class RegistreringTilstandService(private val registreringTilstandRepository: RegistreringTilstandRepository) {
    fun oppdaterRegistreringTilstand(command: OppdaterRegistreringTilstandCommand) {
        val original = registreringTilstandRepository.hentRegistreringTilstand(command.id)
        val oppdatert = original.oppdaterStatus(command.status)
        registreringTilstandRepository.oppdater(oppdatert)
    }

    fun finnRegistreringTilstandMed(status: Status?): List<RegistreringTilstand> {
        return registreringTilstandRepository.finnRegistreringTilstanderMed(status!!)
    }
}